### ISSUE TITLE
ImportDeclaration.parent is not expected to be a TSModuleDeclaration

### ISSUE THREAD

**Issue by flying-sheep** [eslint/typescript-eslint-parser#217]
the rule `import/newline-after-import` accesses a `ImportDeclaration`’s parent, expecting it to [have a `body`](https://github.com/benmosher/eslint-plugin-import/blob/6288cf93e73c3530845609d23fb829c379e357d8/src/rules/newline-after-import.js#L98)

that’s not true if you encounter a `TSModuleDeclaration` → `TSModuleBlock` → `ImportDeclaration`:

```ts
declare module 'postcss-modules' {
	import {Plugin} from 'postcss'
	...
}
```

a fix for this specific problem would be a visitor like `TSModuleDeclaration(m) { m.body = m.statements }`

**Comment by soda0289** [comment]
I have seen this one before. If I remember correctly it doesnt cuase a crash. But it expects that the parent is the Program node, which is true for es6. Not sure if we should update our produced ast nodes or fix the plugin. But in this case it might make sense to use the property body instead of statements. I think class nodes use body property can't remember which nodes, if any, use the statements property.

**Comment by flying-sheep** [comment]
it crashed for me:

```js
const { parent } = node  // parent is basically { type: 'TSModuleDeclaration', statements: [...] }
const nodePosition = parent.body.indexOf(node)  // parent.body === undefined
```

**Comment by soda0289** [comment]
Do you want to submit a PR to rename `statements` property to `body`? I looked in the ESTree spec and they never use statements as a property only body. I think we should follow that same pattern.

**Comment by flying-sheep** [comment]
Done: #218
